test/cpp/end2end/end2end_test.cc

```diff
@@ -291,11 +338,21 @@ class End2endTest : public ::testing::TestWithParam<TestScenario> {
     }
     args.SetString(GRPC_ARG_SECONDARY_USER_AGENT_STRING, "end2end_test");
 
+    auto creators = std::unique_ptr<std::vector<
```

Comment:
So are you adding the interceptors into all the end2end tests? I would advise against that since we should have tests that don't use any interceptor. Let me suggest that you add a new parameter for the number of interceptors and let that either be 0 or non-zero, rather than just having a constant. Let me also suggest that changes in end2end_test should be in a future PR.

Reply:
That makes sense, since the flow for with interceptors and without interceptors is slightly different. I'll revert and do this in a separate PR